Purchase Management: add new cancelation modal

ciprianimike opened the pull request:
### Pemise
The PR https://github.com/Automattic/wp-calypso/pull/67090 is corrupted, I opened this new and addressed the change requests.

P2: pebzTe-6r-p2

* Add new remove-plan-dialog component to the remove purchases page

#### Issue
1039-gh-Automattic/martech

#### Acceptance  Criteria
1039-gh-Automattic/martech

#### Proposed Changes
* Add a new dialog component to show up before the user starts the cancelation flow.

#### Testing Instructions
* Checkout update/pre-cancelation-modal
* Test Calypso locally
* Visit http://calypso.localhost:3000/
* Select a site with an active subscription plan
* Click on te sidebar item Upgrade / Purchases
* Select the plan
* Click on the CTA `Remove <Plan name> plan
* Verify the screenshot image is not blurry

Edge cases:
- Site isComingSoon or private: the modal won't show the site screenshot
- Mobile screen size: the modal won't show the site screenshot
- If the user are cancelling a Jetpack, Professional emails plan: the modal won't show up
- If the user are cancelling a subscription (auto renew on) within the refund window, with a domain, the modal informs them they are going to love the domain and the traffic will be redirected to the related WPCom URL.


#### Before
![before](https://user-images.githubusercontent.com/5706607/188669017-fdcb0005-8683-40db-8255-f8c9bfa970b8.gif)

#### After
https://user-images.githubusercontent.com/5706607/188671098-defa794a-7b57-435c-839c-1d31c1eef619.mp4


#### Issue
Related to 1039-gh-Automattic/martech

ciprianimike committed:
add new cancelation modal

ciprianimike committed:
remove unused fallback translation

ciprianimike committed:
add limits for jetpack plan

ciprianimike committed:
fix Professional Email cancellation

ciprianimike committed:
add plans-list feature for plan + domain

ciprianimike committed:
add plan + domain case list in the cancellation flow

ciprianimike committed:
add domain feature item

dzver reviewed (commented):
The screenshot is blurry or me if width and height props are not provided. Providing width and height resolves the problem.
<img width="476" alt="Screenshot 2022-09-23 at 13 56 13" src="https://user-images.githubusercontent.com/82778/191946213-5c1d1abe-269b-486b-9f08-10805b0d23a6.png">

ciprianimike committed:
set width and height for mshots SiteScreenshot

ciprianimike commented:
> The screenshot is blurry or me if width and height props are not provided. Providing width and height resolves the problem. <img alt="Screenshot 2022-09-23 at 13 56 13" width="476" src="https://user-images.githubusercontent.com/82778/191946213-5c1d1abe-269b-486b-9f08-10805b0d23a6.png">

@dzver I pushed a fix for this.

ciprianimike committed:
fix mShots height

DavidRothstein commented:
I did a quick test and saw that most of the issues from my comment at https://github.com/Automattic/wp-calypso/pull/67090#pullrequestreview-1105154046 are still present (in particular the first, "We seem to have a double confirmation", and the third, "This is being applied to Jetpack too")?

Also noticed that the messaging seems backwards between refundable and non-refundable cancellations:

1. I saw this when cancelling a refundable subscription (where it's actually going to be removed immediately, not "once it expires"):

![refundable](https://user-images.githubusercontent.com/235183/191998376-bf035f69-29d2-47b4-87a1-5b75e06facea.png)

2. And this when cancelling a non-refundable subscription (which is actually going to just turn off auto-renew, not remove anything right away):

![non-refundable](https://user-images.githubusercontent.com/235183/191998394-d0511df8-aad0-4cd9-9df3-af722bb1238a.png)

Seems like it should be the other way around.

Also in the above screenshots, I thought "Best in-class hosting" wasn't supposed to be there anymore (since they aren't losing their hosting)?

And in the first screenshot, "custom domain as primary"... I saw that message even though I don't have a custom domain.  Also, even if I did have one, the wording seems pretty confusing to me.

michaeldcain commented on client/me/purchases/remove-plan-dialog/index.tsx:
```diff
@@ -0,0 +1,162 @@
+							{ domainFeature }
+							{ planFeatures.map( ( feature, index ) => {
+								return (
+									<li key={ index }>
```
Using `index` here is an anti-pattern. You can use `feature` instead: https://github.com/Automattic/wp-calypso/pull/67090#discussion_r965470927

michaeldcain replied:
[Using `index` here is an anti-pattern.](https://github.com/Automattic/wp-calypso/pull/67090#discussion_r965470927) You can use `feature` instead.

michaeldcain commented on client/me/purchases/remove-plan-dialog/index.tsx:
```diff
@@ -0,0 +1,162 @@
+	 * Plan cancellation dialog.
+	 */
+	return (
+		<Dialog
```
This is still missing Tracks events: https://github.com/Automattic/wp-calypso/pull/67090#discussion_r965473304

ciprianimike replied:
@michaeldcain could you please verify the tracking events added here https://github.com/Automattic/wp-calypso/pull/68067/files/80fea556de0d33e45084d1132c5d68428240734b#diff-22f6c96a70e669c1ee86915b052fe607759a7d300568afe8a92d9646c37860d0R33 ?

ciprianimike replied:
@michaeldcain could you please reviews the changes above?

michaeldcain replied:
@ciprianimike: sorry I missed this ping. Those events look good to me.

JessBoctor commented:
👋  From my [initial review](https://github.com/Automattic/wp-calypso/pull/67090#issuecomment-1254028477)

It looks like the issues I raised have all been resolved. 

1. The Blogger plan now has a feature description: 

<img width="532" alt="GH68067-BloggerPlan" src="https://user-images.githubusercontent.com/12505355/192035961-907ad9c0-1ed9-4ce8-af53-7cbb1f2cbff9.png">

2. The images for the public sites are no longer blurry: 

<img width="983" alt="GH68067-BlurryImage" src="https://user-images.githubusercontent.com/12505355/192036036-244397a9-fb95-4079-bf53-061208b619ed.png">

For posterity, I was testing on my 14" MBP A2442. MacOS version 12.5 and Chrome version 105.0.5195.125. Display resolution is set to `Default settings` and the presets to `Apple XDR Display (P3-1600 nits)`. I used an A8c sandbox with local Calypso build for testing.

3. My suggestion about the link for `And many more` was address by @JanaMW27's [comment](https://github.com/Automattic/wp-calypso/pull/67090#issuecomment-1255068751).

spraveenitpro commented:
### For personal plan:

![9m1nxP.png](https://user-images.githubusercontent.com/552016/192048184-93dae209-e2db-4ac9-b8e5-5586c121af31.png)

### For premium plan:

![ZgnAw5.png](https://user-images.githubusercontent.com/552016/192047107-f3ac26f8-8a6d-4a10-b84b-77b360c782fa.png)


### For business plan:

![dhT0S0.png](https://user-images.githubusercontent.com/552016/192048486-93ceb64c-abf2-4ad9-998b-efbe18ce91b6.png)

JanaMW27 commented:
We need the wording: If you cancel your plan, once it expires, you will lose:
In all the modals but the one in the refund window
the: once it expires, is important

ciprianimike committed:
fix using index as key, anti-pattern

ciprianimike committed:
add tracking event on remove plan and cancel action

ciprianimike committed:
remove "best-in-class hosting" from cancellation feature list

dzver commented:
I just want to confirm, is this step supposed to be visible as a second step after clicking the Cancel subscription and refund? (the PR needs rebase, these buttons are now with a different color and copy)

<img width="1085" alt="Screenshot 2022-09-29 at 10 30 47" src="https://user-images.githubusercontent.com/82778/192968340-5b9e0a24-de6a-4261-826c-c489e3b4815b.png">

CC @JanaMW27

ciprianimike commented:
> I just want to confirm, is this step supposed to be visible as a second step after clicking the Cancel subscription and refund? (the PR needs rebase, these buttons are now with a different color and copy)
> 
> <img alt="Screenshot 2022-09-29 at 10 30 47" width="1085" src="https://user-images.githubusercontent.com/82778/192968340-5b9e0a24-de6a-4261-826c-c489e3b4815b.png">
> 
> CC @JanaMW27

@dzver  that step will be skipped, I discussed it yesterday with @JanaMW27  and worked on the fix

JanaMW27 commented:
After some scope and edge cases considerations, we have decided to keep this screen ONLY for users within the refund window.
The rest of the users should ONLY see the modal and not this screen.

ciprianimike committed:
fix site-screenshot getSiteLaunchStatus web test error

ciprianimike committed:
change primary domain check

ciprianimike committed:
set subtitle labels

Set subtitle as 'If you cancel your plan, once it expires, you will lose:' in all the modals but the one in the refund window

ciprianimike commented:
> I did a quick test and saw that most of the issues from my comment at [#67090 (review)](https://github.com/Automattic/wp-calypso/pull/67090#pullrequestreview-1105154046) are still present (in particular the first, "We seem to have a double confirmation", and the third, "This is being applied to Jetpack too")?
> 
> Also noticed that the messaging seems backwards between refundable and non-refundable cancellations:
> 
> 1. I saw this when cancelling a refundable subscription (where it's actually going to be removed immediately, not "once it expires"):
> 
> ![refundable](https://user-images.githubusercontent.com/235183/191998376-bf035f69-29d2-47b4-87a1-5b75e06facea.png)
> 
> 2. And this when cancelling a non-refundable subscription (which is actually going to just turn off auto-renew, not remove anything right away):
> 
> ![non-refundable](https://user-images.githubusercontent.com/235183/191998394-d0511df8-aad0-4cd9-9df3-af722bb1238a.png)
> 
> Seems like it should be the other way around.
> 
> Also in the above screenshots, I thought "Best in-class hosting" wasn't supposed to be there anymore (since they aren't losing their hosting)?
> 
> And in the first screenshot, "custom domain as primary"... I saw that message even though I don't have a custom domain. Also, even if I did have one, the wording seems pretty confusing to me.

@DavidRothstein I updated the PR, the extra step will be present as per the comment: https://github.com/Automattic/wp-calypso/pull/68067#issuecomment-1262036285", I updated the checks for the site having primary domain and removed the hosting. I tested Jetpack subscriptions via jurassic-ninja and it should be fine.

DavidRothstein commented:
I did another round of brief testing.

>I tested Jetpack subscriptions via jurassic-ninja and it should be fine.

For me, the modal is still appearing for Jetpack.  Note that you do need to purchase a Jetpack **plan** to see this (or I think they're calling it "bundle" now) -- for example Jetpack Security, as shown in my original screenshot.  A Jetpack product does not have the problem.

>"once it expires"

I'm not seeing this text appear at all anymore (even when cancelling a nonrefundable subscription, where it's supposed to appear)?

>the extra step will be present as per the comment: https://github.com/Automattic/wp-calypso/pull/68067#issuecomment-1262036285

The behavior I see now doesn't seem to match that comment -- the extra step is present for me in both cases (not just the refundable case)?

Anyway, I guess I can see the logic of having a two-step confirmation for now (maybe even in both cases).  The two steps do provide different information.  However, there is also a fair amount of duplicate information if you just keep the second screen exactly as is (for example, the "Plan Features" link on the second screen completely duplicates the first screen).  And the biggest thing that feels extremely disjointed is jumping in and out of the modal.  You click to cancel, see a modal, click to continue, jump out of the modal, click to continue, and now go into a different kind of modal (full-screen-style) where you get a survey, etc... Couldn't we have both steps of the confirmation inside the modal instead?

>And in the first screenshot, "custom domain as primary"... the wording seems pretty confusing to me.

It still seems confusing to me.  The full context here being:

>If you cancel your plan, you will lose:
>-Your custom domain as primary. Your traffic will be redirected to example.wordpress.com

I don't think most people will understand what that means.  Whereas the current wording that this is replacing (https://github.com/Automattic/wp-calypso/pull/39909 shows an almost-accurate screenshot) uses more basic terminology.  Couldn't we use something more like that?  Perhaps:

>If you cancel your plan, you will lose:
>-The ability to have **example.com** as your primary site address. **example.wordpress.com** will be the new address that people see when they visit your site.

It's longer, but this is a complicated and important concept, and it should be OK to explain it using enough words to make it clear.

ciprianimike committed:
exclude Jetpacks plans and fix copy

ciprianimike commented:
> For me, the modal is still appearing for Jetpack. Note that you do need to purchase a Jetpack **plan** to see this (or I think they're calling it "bundle" now) -- for example Jetpack Security, as shown in my original screenshot. A Jetpack product does not have the problem.

I pushed a fix, before only Jetpack products were excluded, now alsp Jetpack plans.

ciprianimike commented:
> > "once it expires"
> 
> I'm not seeing this text appear at all anymore (even when cancelling a nonrefundable subscription, where it's supposed to appear)?
> 
> > the extra step will be present as per the comment: [#68067 (comment)](https://github.com/Automattic/wp-calypso/pull/68067#issuecomment-1262036285)
> 
> The behavior I see now doesn't seem to match that comment -- the extra step is present for me in both cases (not just the refundable case)?

This text should be now visible only for all modals but subscriptions in the refund window.

ciprianimike committed:
support primaruDomain on remove-plan-dialog and remove-purchase

ciprianimike committed:
support primaryDomain for subscriptions

ciprianimike committed:
support primaryDomain for subscriptions

ciprianimike committed:
update cancelling modal primary domain copy

ciprianimike commented:
> > And in the first screenshot, "custom domain as primary"... the wording seems pretty confusing to me.
> 
> It still seems confusing to me. The full context here being:
> 
> > If you cancel your plan, you will lose:
> > -Your custom domain as primary. Your traffic will be redirected to example.wordpress.com
> 
> I don't think most people will understand what that means. Whereas the current wording that this is replacing (#39909 shows an almost-accurate screenshot) uses more basic terminology. Couldn't we use something more like that? Perhaps:
> 
> > If you cancel your plan, you will lose:
> > -The ability to have **example.com** as your primary site address. **example.wordpress.com** will be the new address that people see when they visit your site.
> 
> It's longer, but this is a complicated and important concept, and it should be OK to explain it using enough words to make it clear.

The wording has been updated, but the longer text require some CSS tuning.

DavidRothstein commented on client/me/purchases/remove-plan-dialog/index.tsx:
```diff
@@ -0,0 +1,182 @@
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const subTitle = ! isRefundable
```
`isRefundable` is a function, but you're using it like a variable -- that may be why this code isn't working correctly?

ciprianimike replied:
<img width="483" alt="Screenshot 2022-10-03 at 15 54 28" src="https://user-images.githubusercontent.com/5706607/193595256-8d65289c-c4a0-4edd-b6e5-257a5f102e96.png">

The domain can be much longer than example.wordpress.com, for long slugs and domains the final look isn't optimal, how about adding a tooltip as:

Your **primary site address ⓘ**. Visitors to your site will see your **wordpress.com address ⓘ** instead.

where ⓘs are tooltips.

JanaMW27 replied:
Could we do: busines...main.wordpress.com?
Or business...wordpres.com?

ciprianimike replied:
@JanaMW27 ok, this could be a good compromise

ciprianimike replied:
<img width="503" alt="Screenshot 2022-10-03 at 16 29 22" src="https://user-images.githubusercontent.com/5706607/193602963-e631d11f-df8e-4718-a365-33afefd387fa.png">

@JanaMW27  @DavidRothstein  This is how it looks cutting the domain names, going down from 5 to 3 rows

DavidRothstein replied:
Similar to what @JanaMW27 said below, I wouldn't recommend truncating their custom domain.

If you want to truncate the `.wordpress.com` domain, I would at least do it at a significantly longer word length?  I think it can be a fair amount longer than you have it now without looking terrible.  And at that point, we'd also be talking about a very tiny number of sites, so I'm not sure how much it really matters... (For what it's worth, you could also just use `overflow-wrap` CSS to keep it from looking like a complete disaster and then not worry about the rest.)

ciprianimike replied:
All right, I'm doing it

DavidRothstein reviewed (commented):
Thanks for the fixes!

I'm now seeing the opposite behavior for "once it expires", though -- I'm seeing it in both cases (refundable and nonrefundable).  I left an inline comment about a possible cause.

I'm also still seeing the second confirmation screen in both cases.

Thanks for trying out my wording changes for the domain.  Looking at your screenshot, I am not sure why it couldn't be a list item (with the red X) though?  Also agree the text looks a little long.  And it might help to bold the domain names.

I tried shortening it a bit, something like this:

>If you cancel your plan, you will lose:
>- **example.com** as your primary site address. Visitors to your site will see **example.wordpress.com** as the address instead

Rough mockup:

![primary-domain-removal](https://user-images.githubusercontent.com/235183/193349024-cea902ec-cf43-4829-a996-9ffca7045cd0.png)

Will link to this discussion at pebzTe-gw-p2 also.

ciprianimike committed:
update domains copy and refundable window logic

JanaMW27 commented:
Could we force it to not truncate the domain name?

Sent from my iPhone

> On 3 Oct 2022, at 16:30, Michele Cipriani ***@***.***> wrote:
> 
> ﻿
> @ciprianimike commented on this pull request.
> 
> In client/me/purchases/remove-plan-dialog/index.tsx:
> 
> > +	wpcomURL,
> +}: RemovePlanDialogProps ) => {
> +	const translate = useTranslate();
> +	const siteName = site.name ?? '';
> +
> +	/**
> +	 * Istantiate site's plan variables.
> +	 */
> +	const productSlug = site.plan?.product_slug;
> +	const planLabel = site.plan?.product_name_short;
> +	const isComingSoon = site.is_coming_soon;
> +	const isPrivate = site.is_private;
> +	const launchedStatus = site.launch_status === 'launched' ? true : false;
> +	const shouldUseSiteThumbnail =
> +		isComingSoon === false && isPrivate === false && launchedStatus === true;
> +	const subTitle = ! isRefundable
> 
> 
> @JanaMW27 @DavidRothstein This is how it looks cutting the domain names
> 
> —
> Reply to this email directly, view it on GitHub, or unsubscribe.
> You are receiving this because you were mentioned.

DavidRothstein commented on client/me/purchases/remove-purchase/index.jsx:
```diff
@@ -212,6 +226,36 @@ class RemovePurchase extends Component {
+				? primaryDomainName
+				: primaryDomainName.substr( 0, 10 ) + '…' + primaryDomainName.slice( -10 );
+		const wordpressComURL =
+			site.wpcom_url.legth < 30
```
Typo (`legth`)

DavidRothstein commented on client/me/purchases/manage-purchase/index.jsx:
```diff
@@ -500,6 +516,34 @@ class ManagePurchase extends Component {
+						hasDomain={ customDomain }
+						isRefundable={ isRefundable( purchase ) }
+						primaryDomain={ primaryDomainName }
+						wpcomURL={ site.wpcom_url }
```
If you're going to do URL truncation, you should do it here too.

ciprianimike committed:
fix typo

ciprianimike committed:
fix long domain names with CSS and trunkating wpcom URLs

ciprianimike commented:
I worked on the CSS having keeping the full domain name and truncating the wpcom URL only if it take more than a line.
<img width="487" alt="Screenshot 2022-10-04 at 13 41 53" src="https://user-images.githubusercontent.com/5706607/193811113-a47ae97c-5fd3-48b7-918a-d89ad139dc24.png">

ciprianimike committed:
skip extra step by removing purchase in manage-purchase page

ciprianimike committed:
show CancelForm for non refundable subscriptions

ciprianimike commented:
@DavidRothstein all your feedbacks should be addressed, could you please review? and thanks for your big patience

DavidRothstein commented on client/me/purchases/remove-plan-dialog/index.tsx:
```diff
@@ -0,0 +1,205 @@
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const subTitle = isRefundable
+		? translate( 'If you cancel your plan, once it expires, you will lose:' )
+		: translate( 'If you cancel your plan, you will lose:' );
```
These are still backwards.  But otherwise, this part is now working correctly.

DavidRothstein replied:
Actually, not entirely sure `isRefundable` is the correct thing to check on its own -- if the subscription isn't refundable but auto-renew is off, this dialog will still result in the plan being removed immediately, I think (not when it expires)?  I haven't traced this code though.

ciprianimike replied:
could you please elaborate this? `isRefundable` is a bool prop now in the `RemovePlanDialog` componente. The value is calculated by using the function `isRefundable()` in the parent components.

DavidRothstein replied:
I think what I'm saying is that the following is the text/behavior I would expect:
- **If the subscription is refundable:**  Text should say "If you cancel your plan, you will lose" and the end result is that the subscription should be removed immediately
- **If the subscription is not refundable and auto-renew is on:**  Text should say "If you cancel your plan, once it expires, you will lose" and the end result is that the subscription should have auto-renew turned off
- **If the subscription is not refundable and auto-renew is off:**  Text should say "If you ~cancel~ [or perhaps "remove", to match the button text they click on to get into this dialog?] your plan, you will lose" and the end result is that the subscription should be removed immediately

At least I'm pretty sure that's what it should be.

ciprianimike replied:
this logic has been implemented along with the requested copy

DavidRothstein reviewed (commented):
I tried it out briefly, and the "Cancel plan" link now seems to always be removing the plan **immediately** (even when outside the refund window and auto-renew is on, which is where it previously just turned auto-renew off).

Otherwise looked pretty good to me, although the double-confirmation with modal for step 1 and non-modal for step 2 is still there...

The domain changes look good now, but I did see this layout issue (which I hadn't seen before - it may be new):

![cancel-plan](https://user-images.githubusercontent.com/235183/193896947-61b477ac-5964-4824-b621-a9f71c07ef01.png)

I'll be AFK for the next day or so, so someone else may have to pick up any reviews.

ciprianimike commented:
> I tried it out briefly, and the "Cancel plan" link now seems to always be removing the plan **immediately** (even when outside the refund window and auto-renew is on, which is where it previously just turned auto-renew off).
> 
> Otherwise looked pretty good to me, although the double-confirmation with modal for step 1 and non-modal for step 2 is still there...
> 
> The domain changes look good now, but I did see this layout issue (which I hadn't seen before - it may be new):
> 
> ![cancel-plan](https://user-images.githubusercontent.com/235183/193896947-61b477ac-5964-4824-b621-a9f71c07ef01.png)
> 
> I'll be AFK for the next day or so, so someone else may have to pick up any reviews.

These feedbacks are clear, I'm testing and fixing it

ciprianimike committed:
css fixes --with-domain-feature

ciprianimike commented:
CSS issues has been fixed

ciprianimike committed:
fix subscription cancellation cancellation flow

ciprianimike committed:
fix subscription cancellation logic and copy

ciprianimike commented:
@DavidRothstein all your feedback should be arranged, please check this PR once again

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7568066

Thank you @ciprianimike for including a screenshot in the description! This is really helpful for our translators.

ciprianimike committed:
prevent new cancellation modal to show up for non english langauges

DavidRothstein commented on client/me/purchases/remove-plan-dialog/index.tsx:
```diff
@@ -0,0 +1,213 @@
+	} else if ( ! isRefundable && isAutoRenewing ) {
+		subTitle = translate( 'If you cancel your plan, once it expires, you will lose:' );
+	} else if ( ! isRefundable && ! isAutoRenewing ) {
+		translate( 'If you remove your plan, you will lose:' );
```
This is not being assigned to `subTitle` -- however, I'm not sure that's the only problem here, since I'm not seeing the subtitle displayed in any other scenario either.  This is what I see no matter what I try:

![cancel-plan](https://user-images.githubusercontent.com/235183/194682364-ef19b005-1139-4f79-bd8f-c09e93a72345.png)

ciprianimike replied:
> Not sure if I'm doing something wrong, but I'm still seeing problems for a non-refundable subscription with auto-renew on:
> 
> * Going through the cancellation flow still results in the plan being removed rather than auto-renew being turned off.
> * I also noticed the "Back" button inside the cancellation survey was broken in this case (clicking it did nothing).
> 
> In addition, the intro paragraph never seems to be displaying in the modal currently (see inline comment for more about that).
> 

I fixed the auto-renew logic and the 3 bugs above look resolved to me.

> Besides the above, it looks like it's mostly working, but the jumping in and out of different modals while cancelling still seems disjointed to me. @Automattic/shilling should perhaps take a closer look at the overall flow next week. I'd post screenshots or a video but don't have time right now.

I'm progressing on the work for this case: pebzTe-f6-p2 it will be covered by a different PR and this should be the only disjointed case. For all the other cases the Cancellation Modal shows the survey form without disjunctions.

DavidRothstein reviewed (commented):
Not sure if I'm doing something wrong, but I'm still seeing problems for a non-refundable subscription with auto-renew on:
- Going through the cancellation flow still results in the plan being removed rather than auto-renew being turned off.
- I also noticed the "Back" button inside the cancellation survey was broken in this case (clicking it did nothing).

In addition, the intro paragraph never seems to be displaying in the modal currently (see inline comment for more about that).

Besides the above, it looks like it's mostly working, but the jumping in and out of different modals while cancelling still seems disjointed to me.  @Automattic/shilling should perhaps take a closer look at the overall flow next week.  I'd post screenshots or a video but don't have time right now.

ciprianimike committed:
update subtitle logic

ciprianimike committed:
update cancel flow type logic

ciprianimike committed:
remove ´getLocaleSlug() === 'en'´ check since all the strings were translated

ciprianimike committed:
fix state when closeDialog for subscriptions

DavidRothstein commented:
I tested again and am still seeing some of the previously-mentioned problems, in particular:
- For a non-refundable subscription with auto-renew on, going through the cancellation flow still results in the plan being removed rather than auto-renew being turned off.
- Although the intro paragraph is now displaying in the modal (that part is fixed) I'm seeing the wrong message in the case of a removable subscription: "If you cancel your plan, once it expires, you will lose" rather than "If you remove your plan, you will lose".  It looks like issue I mentioned before (where the translated string is not being assigned to the `subTitle` variable) was not fixed.

>>the jumping in and out of different modals while cancelling still seems disjointed to me.
>
>I'm progressing on the work for this case: pebzTe-f6-p2 it will be covered by a different PR and this should be the only disjointed case. For all the other cases the Cancellation Modal shows the survey form without disjunctions.

It's currently happening for any plan cancellation within the refund window (the link you provided is for the case where there's a plan and domain, which is pretty common in its own right, but is not the only case).

Here's a video showing the behavior for cancelling just a plan (no domain involved):

https://user-images.githubusercontent.com/235183/195505670-4c7ad85a-3722-4e32-bdad-5efd7cdc2325.mp4

What is the rationale for fixing that flow as a followup, rather than launching this with the intended experience in place to begin with? If the goal is to get something out sooner, would it be better to just release the modal for the **non-refundable** case only to start (and then hold off on the refundable case until it's actually ready)?

Regarding the non-refundable case, this pull request is indeed replacing the old screen with the new modal, which is good.   Comparing them below, I do see two things that have gone missing that may be important:
1. The "Ask a Happiness Engineer" link -- do we really want to remove that?  I would guess it is useful to give people an easy way to contact us if they're thinking of cancelling.  And it seems like it would be pretty easy to put that in the modal.
2. The "subscription will be removed on October 12, 2023" text, although not critical, seems like it provides useful information that allows the person to understand what will happen.  Do we want to consider changing the text in the modal to something like: "If you cancel your plan, once it expires (on October 12, 2023) you will lose"?

**Old cancel confirmation:**

![old-cancel-page](https://user-images.githubusercontent.com/235183/195505693-5495cf94-250a-4333-a61f-d965d65bda58.png)

**New cancel confirmation:**

![new-cancel-modal](https://user-images.githubusercontent.com/235183/195505690-cbaf0e3e-59aa-4e65-b319-7c486b99d06a.png)

michaeldcain commented:
This is being broken down into smaller diffs per this post: pebzTe-oH-p2

I'm going to make this a draft PR so that we'll get pinged if it ever gets changed back.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7568066).
